src/ml/concentration_svm_classifier.cpp

```diff
@@ -0,0 +1,49 @@
+#include <algorithm>
+#include <cmath>
+#include <stdlib.h>
+#include <iostream>
+#include "brainflow_constants.h"
+#include "concentration_svm_classifier.h"
+#include "get_dll_dir.h"
+
+int ConcentrationSVMClassifier::prepare ()
+{
+    char path[1024];
+    bool res = get_dll_path (path);
+    char * fullPath = new char [std::strlen(path)+std::strlen("brainflow_svm.model")+1];
+    std::strcpy (fullPath,path);
+    std::strcat (fullPath ,"brainflow_svm.model");
+    model = svm_load_model (fullPath);
+    return (int)BrainFlowExitCodes::STATUS_OK;
+}
+
+int ConcentrationSVMClassifier::predict (double *data, int data_len, double *output)
+{
+    if ((data_len < 5) || (data == NULL) || (output == NULL))
+    {
+        safe_logger (spdlog::level::err, "Classifier has already been prepared.");
+        return (int)BrainFlowExitCodes::INVALID_BUFFER_SIZE_ERROR;
+    }
+    struct svm_node *x;
+    x =  (struct svm_node *) malloc(data_len*sizeof(struct svm_node));
```

Comment:
looks like memory leak, I dont see free call in the end. You can try to run it with valgrind to test

Reply:
and clang-format is definitely off on your PC